dm/dm/master/server.go

```diff
@@ -573,6 +573,18 @@ func (s *Server) OperateTask(ctx context.Context, req *pb.OperateTaskRequest) (*
 
 	resp.Result = true
 	resp.Sources = s.getSourceRespsAfterOperation(ctx, req.Name, sources, []string{}, req)
+
+	if expect == pb.Stage_Stopped {
+		// delete meta data for optimist
+		if len(req.Sources) == 0 {
+			err2 = s.optimist.RemoveMetaDataWithTask(req.Name)
```

Comment:
we should check the resp, make sure worker is really stopped before delete the table

Reply:
Actually we remove the subtask in scheduler at L564, so even worker not really stopped, the meta should be removed.

Reply:
And now master start/stop the task by update etcd, so it only depends on whether put/delete etcd successfully.